### PR TITLE
Add bo blocks

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: ElasticaAdmin/DisplayIcon/ElasticaSearchStrategy.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * Check if the strategy support this block
+     *
+     * @param string $block
+     *
+     * @return boolean
+     */
+    public function support($block)
+    {
+        return 'elastica_search' == $block;
```
Comment (alavieille) [inline ElasticaAdmin/DisplayIcon/ElasticaSearchStrategy.php]: use ===

File: ElasticaAdmin/GenerateForm/ElasticaSearchStrategy.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class ElasticaSearchStrategy extends AbstractBlockStrategy
+{
+    /**
+     * @param BlockInterface $block
+     *
+     * @return bool
+     */
+    public function support(BlockInterface $block)
+    {
+        return 'elastica_search' == $block->getComponent();
```
Comment (alavieille) [inline ElasticaAdmin/GenerateForm/ElasticaSearchStrategy.php]: use ===

File: ElasticaAdminBundle/Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+open_orchestra_backoffice:
+    block:
+        elastica_search:
+            title: Elastica search block
+
+open_orchestra_elastica_admin:
+    block:
+        elastica_search:
+            title: Elastica search
+            description: Elastica search field
```
Comment (alavieille) [inline ElasticaAdminBundle/Resources/translations/messages.en.yml]: eof

File: ElasticaAdminBundle/Resources/translations/messages.fr.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+open_orchestra_elastica_admin:
+    block:
+        elastica_search:
+            title: Recherche elastica
+            description: Champs de recherche elastica
```
Comment (alavieille) [inline ElasticaAdminBundle/Resources/translations/messages.fr.yml]: eof

File: ElasticaAdmin/DisplayBlock/ElasticaSearchStrategy.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Check if the strategy support this block
+     *
+     * @param ReadBlockInterface $block
+     *
+     * @return boolean
+     */
+    public function support(ReadBlockInterface $block)
+    {
+        return 'elastica_search' === $block->getComponent();
```
Comment (ngilain) [inline ElasticaAdmin/DisplayBlock/ElasticaSearchStrategy.php]: Why don't use a class constant for the name as on the other blocks ?

File: ElasticaAdmin/DisplayIcon/ElasticaSearchStrategy.php (hunk 1 above)
Comment (ngilain) [inline ElasticaAdmin/DisplayIcon/ElasticaSearchStrategy.php]: Same remark about the class constant

File: ElasticaAdmin/GenerateForm/ElasticaSearchStrategy.php (hunk 2 above)
Comment (ngilain) [inline ElasticaAdmin/GenerateForm/ElasticaSearchStrategy.php]: and again :)

File: ElasticaAdmin/DisplayBlock/ElasticaSearchStrategy.php (hunk 5 above)
Comment (nicolasThal) [inline ElasticaAdmin/DisplayBlock/ElasticaSearchStrategy.php]: That was going to be one of my first refacto :).

To be sure of the constant position, I wanted to finish the Front part

**Comment by nicolasThal** [comment]
@itkg-ngilain @itkg-alavieille pr updated

**Pull request merged**